NDVItimeseries.R

```diff
@@ -0,0 +1,527 @@
+NDVIadmin3 <- dplyr::left_join(NDVI1admin3, NDVI2admin3)
+
+# geospatial
+nigerMapAdmin2 <- 
+  st_read("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/Level 2 (1)/wb_niger_admin2_shapefile/niger_admin2.shp") 
+
+
+nigerMapAdmin3 <- 
+  st_read("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/Level 3 (1)/NER_adm03_feb2018.shp")
+
+
+# Don't show scientific notation
+options(scipen = 999)
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Clean titles and data -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("_NDVI", "", colnames(NDVIadmin2)[col])
+}
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("X", "", colnames(NDVIadmin2)[col])
+}
+
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("_NDVI", "", colnames(NDVIadmin3)[col])
+}
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("X", "", colnames(NDVIadmin3)[col])
+}
+head(NDVIadmin3)
+
+
+# Convert data from Wide to Long ---- 
+ndviData2 <- 
+  NDVIadmin2 %>%
+  select(admin1Name, admin1Pcod, 
+         admin2Name, admin2Pcod, "19810701": "20191231") 
+
+ndviDataLong2 <- 
+  gather(ndviData2, 
+         key = Date, 
+         value = ndvi, 
+         "19810701":"20191231") %>% 
+  mutate(Date = ymd(Date)) #convert from char to dates
+# "19810701":"20191231") 
+
+ndviDataLong2$ndvi <- as.numeric(ndviDataLong2$ndvi) 
+ndviDataLong2 <- ndviDataLong2 %>% 
+  mutate(newNDVI = 0.0001 *ndvi)
+
+ndviData3 <-
+  NDVIadmin3 %>%
+  select(adm_03, rowcacode3,  
+         "19810701":"20191231")
+
+ndviDataLong3 <-
+  gather(ndviData3,
+         Date,
+         ndvi,
+         "19810701":"20191231") %>% 
+  mutate(Date = ymd(Date)) # Create new col with dates instead of chars
+
+ndviDataLong3$ndvi <- as.numeric(ndviDataLong3$ndvi) 
+ndviDataLong3 <- ndviDataLong3 %>% 
+  mutate(newNDVI = 0.0001 *ndvi)
+
+data_admin2 <- 
+  ndviDataLong2 %>%
+  mutate(year = year(Date), 
+         month = month(Date), 
+         day = day(Date)) %>%
+  ungroup()
+
+data_admin3 <- 
+  ndviDataLong3 %>%
+  mutate(year = year(Date), 
+         month = month(Date), 
+         day = day(Date)) %>%
+  ungroup()
+
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Generate Total NDVI Graphics By Admin 0 (country) and Admin 1 (building off admin 2) ---- 
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Admin 0 (one line for the entire region)
+peakndvi <- 
+  yearData_admin2 %>%
+  group_by(year) %>%
+  mutate(meanAnnualndvi = mean(peak_ndvi)) %>%
+  ungroup() %>%
+  distinct(year, meanAnnualndvi)
+
+peakndvi %>% 
+  ggplot(aes(x=year, y=meanAnnualndvi)) +
+  geom_line() + 
+  labs(title="Peak NDVI by Year", x="Year", y ="Peak NDVI") +
```

Comment:
this is over the entire country - recommend labeling as such (also hard to think about what that might mean to have peak ndvi over the entire country. smaller units generally are more interpretable as here you will be mixing in urban areas, desert, and ag areas)